disaggregated_memory/memories/external_remote_memory_v2.py

```diff
@@ -0,0 +1,215 @@
+
+        # We will either use size or addr range. This variable is used to keep
+        # a track of that.
+        self._set_using_addr_ranges = False
+
+        # The ExternalMemory is an AbstractMemory object that connects
+        # gem5 to SST as an external memory.
+        self.outgoing_request_bridge = ExternalMemory()
+
+        # Indicate whether the user is using SST or not.
+        self.outgoing_request_bridge.use_sst_sim = use_sst_sim
+
+        # TODO: The range and physical_address_ranges should have the same name
+        # to avoid confusion. The address map needs to be visible to the cores
+        # to use all types of CPUs including the O3 CPU.
+        self.outgoing_request_bridge.in_addr_map = True
+
+        # The user needs to provide either the size of the remote memory or the
+        # range of the remote memory.
+        if size is None and addr_range is None:
+            fatal("External memory needs to either have a size or a range!")
+        else:
+            if addr_range is not None:
+                self.outgoing_request_bridge.physical_address_ranges = [
+                    addr_range
+                ]
+                self._size = (
+                    self.outgoing_request_bridge.physical_address_ranges[
+                        0
+                    ].size()
+                )
+                self._set_using_addr_ranges = True
+            # The size will be setup in the board in case ranges are not given
+            # by the user.
+            else:
+                # There is no range information provided by the user. Depending
+                # upon the ISA, we have to fix the address.
+                # TODO: There is no way for the AbstractMemorySystem to know
+                # that ISA is board is using.
+                warn(
+                    "The ExternalMemory interface is set using a size. "
+                    + "Defaulting to 0x80000000 (ARM/RISCV) style start"
+                    + "address. The program may crash if you're using X86."
+                )
+                self.outgoing_request_bridge.physical_address_ranges = [
+                    AddrRange(start=0x80000000, size=size)
+                ]
+                self._size = (
+                    self.outgoing_request_bridge.physical_address_ranges[
+                        0
+                    ].size()
+                )
+
+        # TODO: Marked for deletion
+        # If there is a remote latency specified, create a non_coherent
+        # cross_bar.
+        # if link_latency > 0:
+        #     self._xbar_required = True
+        #     self._link_latency = link_latency
+
+    def get_size(self):
+        return self._size
+
+    def is_xbar_required(self):
+        # If an XBar is required, it should be added in the connect_things to
+        # avoid initializing an orphan node.
+        raise NotImplementedError("This is a deprecated method for v2")
+        return self._xbar_required
```

Comment:
delete

Reply:
This is completely deleted now! Even the methods are gone.